src/main/java/DukeStorage.java

```diff
@@ -0,0 +1,133 @@
+        return this.taskArrayList;
+    }
+
+    private Task getTaskFromDatabase(String taskString) {
+        String[] currentTaskSplit = taskString.split(" [|] ");
+        String taskIndicator = currentTaskSplit[0];
+        String taskIsCompleted = currentTaskSplit[1];
+        String taskDescription = currentTaskSplit[2];
+
+        switch (taskIndicator) {
+            case TASK_IS_TODO:
+                Todo todoReturn = new Todo(taskDescription);
+                if (taskIsCompleted.equals(TASK_COMPLETED_INT)) {
+                    todoReturn.setTaskAsDone();
+                }
+                return todoReturn;
+            case TASK_IS_DEADLINE:
+                String deadlineDate = currentTaskSplit[3];
+                Deadline deadlineReturn = new Deadline(taskDescription, deadlineDate);
+                if (taskIsCompleted.equals(TASK_COMPLETED_INT)) {
+                    deadlineReturn.setTaskAsDone();
+                }
+                return deadlineReturn;
+            case TASK_IS_EVENT:
+                String eventDate = currentTaskSplit[3];
+                Event eventReturn = new Event(taskDescription, eventDate);
+                if (taskIsCompleted.equals(TASK_COMPLETED_INT)) {
+                    eventReturn.setTaskAsDone();
+                }
+                return eventReturn;
+        }
+        return null;
+    }
+
+    private void createNewDataDirectory() {
+        new File("../data").mkdirs();
+        try {
+            new FileWriter("../data/duke.txt");
+        } catch (IOException e) {
+            System.out.println("Unable to create duke.txt file in data directory");
+        }
+    }
+
+    private String getTaskCompletionInInt(Task task) {
+        String taskCompletionInteger;
+        if (task.isCompleted()) {
+            taskCompletionInteger = TASK_COMPLETED_INT;
+        } else {
+            taskCompletionInteger = TASK_NOT_COMPLETED_INT;
+        }
+        return taskCompletionInteger;
+    }
+
+    private String cleanTaskToSave(Task task) {
+        String taskStringToSave = "";
+        if (task instanceof Todo) {
+            Todo taskTodo = (Todo) task;
+            taskStringToSave = TASK_IS_TODO + " | " +
+                    getTaskCompletionInInt(taskTodo) + " | " +
+                    taskTodo.getTaskDescription();
+        } else if (task instanceof Event) {
+            Event taskEvent = (Event) task;
+            taskStringToSave = TASK_IS_EVENT + " | " +
+                    getTaskCompletionInInt(task) + " | " +
+                    taskEvent.getTaskDescription() + " | " +
+                    taskEvent.getEventDate();
+        } else if (task instanceof Deadline) {
+            Deadline taskDeadline = (Deadline) task;
+            taskStringToSave = TASK_IS_DEADLINE + " | " +
+                    getTaskCompletionInInt(task) + " | " +
+                    taskDeadline.getTaskDescription() + " | " +
+                    taskDeadline.getDeadlineDate();
+        }
```

Comment:
This is missing an`else` statement. Refer to the standard [if-else layout](https://se-education.org/guides/conventions/java/basic.html#layout).